### PR TITLE
fix(45224): No inlay hints for setter parameter

### LINKED ISSUE #45224: No inlay hints for setter parameter
```ts
class Foo {
    #value = 42;
    get foo(): number { return this.#value; }
    set foo(value) { this.#value = value; }
}
```

**Expected**: Inlay hint for `value` as `: number`
**Actual**: No inlay hint.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #45224

### CODE REVIEW

File: src/services/inlayHints.ts [hunk 1]
```diff
@@ -64,8 +64,8 @@ namespace ts.InlayHints {
                 visitCallOrNewExpression(node);
             }
             else {
-                if (preferences.includeInlayFunctionParameterTypeHints && isFunctionExpressionLike(node)) {
-                    visitFunctionExpressionLikeForParameterType(node);
+                if (preferences.includeInlayFunctionParameterTypeHints && (isArrowFunction(node) || isFunctionExpression(node) || isSetAccessorDeclaration(node))) {
```
Comment (DanielRosenwasser) [inline src/services/inlayHints.ts:67]: Maybe an aside from this PR - it feels like anything that can be contextually typed should get this treatment. Basically whatever's in `hasContextSensitiveParameters`, right?

https://github1s.com/microsoft/TypeScript/blob/8f6ad87d9980d028b312924c9bca1ff92ddb02a8/src/compiler/checker.ts#L16395-L16399

**Review by DanielRosenwasser — APPROVED** [review]

File: src/services/inlayHints.ts (hunk 1 above)
Comment (a-tarasyuk) [reply src/services/inlayHints.ts]: @DanielRosenwasser  I've added condition with `hasContextSensitiveParameters`

**Comment by DanielRosenwasser** [comment]
@typescript-bot perf test this

**Comment by DanielRosenwasser** [comment]
If you resolve the conflicts I should be able to merge this in.

**Pull request merged**